azurerm/data_source_storage_account_sas_test.go

```diff
@@ -0,0 +1,277 @@
+package azurerm
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"testing"
+	"time"
+)
+
+func TestAccDataSourceArmStorageAccountSasRead(t *testing.T) {
+	dataSourceName := "data.azurerm_storage_account_sas.test"
+	rInt := acctest.RandInt()
+	rString := acctest.RandString(4)
+	location := testLocation()
+	utcNow := time.Now().UTC()
+	startDate := utcNow.Format(time.RFC3339)
+	endDate := utcNow.Add(time.Hour * 24).Format(time.RFC3339)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAzureRMStorageAccountSas_basic(rInt, rString, location, startDate, endDate),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(dataSourceName, "https_only", "true"),
+					resource.TestCheckResourceAttr(dataSourceName, "start", startDate),
+					resource.TestCheckResourceAttr(dataSourceName, "expiry", endDate),
+					resource.TestCheckResourceAttrSet(dataSourceName, "sas"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataSourceAzureRMStorageAccountSas_basic(rInt int, rString string, location string, startDate string, endDate string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name = "acctestsa-%d"
+  location = "%s"
+}
+
+resource "azurerm_storage_account" "test" {
+  name = "acctestsads%s"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+
+  location = "${azurerm_resource_group.test.location}"
+  account_tier = "Standard"
+  account_replication_type = "LRS"
+
+  tags {
+    environment = "production"
+  }
+}
+
+data "azurerm_storage_account_sas" "test" {
+  connection_string = "${azurerm_storage_account.test.primary_connection_string}"
+  https_only        = true
+  resource_types {
+    service   = true
+    container = false
+    object    = false
+  }
+  services {
+    blob  = true
+    queue = false
+    table = false
+	file  = false
+  }
+  start   = "%s"
+  expiry  = "%s"
+  permissions {
+	read    = true
+	write   = true
+	delete  = false
+    list    = false
+    add     = true
+	create  = true
+	update  = false
+	process = false
```

Comment:
_minor_ can we make the spacing consistent here?

Reply:
I see what happened. Go fmt didn't change the tabs/spaces in the string literal.

Will fix this in a minute.